diff -pruN mpd-0.20.9/src/ls.cxx mpd-0.20.9.2/src/ls.cxx
--- mpd-0.20.9/src/ls.cxx	2017-05-08 13:15:49.000000000 +0000
+++ mpd-0.20.9.2/src/ls.cxx	2017-07-16 16:13:06.772000000 +0000
@@ -42,6 +42,14 @@ static const char *const remoteUrlPrefix
 	"mmsu://",
 #endif
 #ifdef ENABLE_FFMPEG
+	"http://",
+	"https://",
+	"mms://",
+	"mmsh://",
+	"mmst://",
+	"mmsu://",
+	"tcp://",
+	"udp://",
 	"gopher://",
 	"rtp://",
 	"rtsp://",
