diff -pruN exfat/Makefile.am exfat2/Makefile.am
--- exfat/Makefile.am	2017-07-09 21:49:35.600000000 +0000
+++ exfat2/Makefile.am	2017-07-09 21:50:37.172000000 +0000
@@ -20,4 +20,4 @@
 #	51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
-SUBDIRS = libexfat dump fsck fuse label mkfs
+SUBDIRS = libexfat fsck label mkfs
diff -pruN exfat/configure.ac exfat2/configure.ac
--- exfat/configure.ac	2017-07-09 21:49:35.600000000 +0000
+++ exfat2/configure.ac	2017-07-09 21:50:15.792000000 +0000
@@ -31,13 +31,10 @@ AC_PROG_CC_C99
 AC_PROG_RANLIB
 AM_PROG_AR
 AC_SYS_LARGEFILE
-PKG_CHECK_MODULES([FUSE], [fuse])
 AC_CONFIG_HEADERS([libexfat/config.h])
 AC_CONFIG_FILES([
 	libexfat/Makefile
-	dump/Makefile
 	fsck/Makefile
-	fuse/Makefile
 	label/Makefile
 	mkfs/Makefile
 	Makefile])
