--- aaa/src/Makefile	2017-05-10 12:27:57.000000000 +0700
+++ bbb/src/Makefile	2017-05-10 12:57:05.467635402 +0700
@@ -1,16 +1,16 @@
 all: clean doyacc dolex markdown.out
 
 doyacc:
-	bison -d -t --locations -omarkdown.y.c markdown.y
+	bison -d -t --locations -omarkdown.y.c markdown.y
 
 dolex:
-	flex -omarkdown.lex.c markdown.lex
+	flex -omarkdown.lex.c markdown.lex
 
 markdown.out: main.o input.o markdown.y.o markdown.lex.o htmltags.o tagstack.o tagstackcheck.o blocknode.o blocknodestack.o strbuf.o
-	gcc $^ -o $@ -g
+	$(CC) $(LDFLAGS) $^ -o $@ -g
 
 ..c.o:
-	gcc -c $< -g
+	$(CC) $(CFLAGS) $< -g
 
 clean:
 	rm -rf *.o 

